### PR TITLE
Servlet60 SessionCookieConfig new APIs

### PR DESCRIPTION
**Pull request by pmd1nh**
https://github.com/OpenLiberty/open-liberty/issues/20417

Support new APIs in jakarta.servlet.SessionCookieConfig

https://jakarta.ee/specifications/servlet/6.0/apidocs/jakarta.servlet/jakarta/servlet/sessioncookieconfig#setAttribute(java.lang.String,java.lang.String)

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build       #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
4 failed tests.  Requested rebuild for these 2 buckets and all tests are good.

com.ibm.ws.session.cache_fat FAT tests | 174 | 0 | 2 |   |  
-- | -- | -- | -- | -- | --
com.ibm.ws.session.cache_fat FAT tests | 182 | 0 | 0 |   |  

com.ibm.ws.session.cache_fat_config FAT tests | 39 | 0 | 2 |   |  
com.ibm.ws.session.cache_fat_config FAT tests | 39 | 0 | 0 |   |

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
docker problem caused lot of failure.  resubmit

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
@zhang-ca  FYI

**Comment by pnicolucci** [comment]
Code review comments:

1) Add tests
2) need translations for the exceptions, in a follow on PR is fine
3) still need to address `cookie-config` in the web.xml
4) Give Allan an FYI so he is aware of the changes to the session component

**Comment by pmd1nh** [comment]
Added FAT

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

**Comment by pmd1nh** [comment]
corrected spelling, removed white spaces...

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat

**Comment by pmd1nh** [comment]
1 failed and 4 errors but they are not related to this session change.

File: dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionCookieConfigImpl.java [hunk 1]
```diff
@@ -203,23 +253,26 @@ public void setSecure(boolean b, boolean externalCall) {
...
+        SessionCookieConfigImpl temp = new SessionCookieConfigImpl(getName(), getDomain(), getPath(), comment, getMaxAge(), isHttpOnly(), isSecure());
+        temp.attributes = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        temp.attributes.putAll(this.attributes);
+
         return temp;
     }
 
-    private void throwWarning() {
+    protected void throwWarning() {
         String msg = nls.getString("programmatic.sessions.already.been.initialized");
```
Comment (pnicolucci) [inline dev/com.ibm.ws.session/src/com/ibm/ws/session/SessionCookieConfigImpl.java:270]: opened: https://github.com/OpenLiberty/open-liberty/issues/22433 to update the nls here and get a translated message.

**Review by pnicolucci — APPROVED** [review]

**Comment by pmd1nh** [comment]
#libby

**Review by zhang-ca — COMMENTED** [review]
Looks fine to me. Thanks

**Pull request merged**